presto-main/src/test/java/com/facebook/presto/operator/TestExchangeClient.java

```diff
@@ -120,6 +120,7 @@ public void testHappyPath()
                 1,
                 new Duration(1, MINUTES),
                 true,
+                false,
```

Comment:
Currently the assync exchange code path is untested. Think of how can you have them both tested. Maybe add a simple unit / integration test. For example you can have this behaviour configured with a session property and add a single test case to the `DistributedQueryRunner`.